integration-tests/src/tests/references.test.ts

```diff
@@ -0,0 +1,78 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2023 Savoir-faire Linux. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as assert from 'assert'
+import * as vscode from 'vscode'
+import path from 'path'
+import { assertWillComeTrue } from '../utils/async'
+
+suite('Bitbake Definition Test Suite', () => {
+  const filePath = path.resolve(__dirname, '../../project-folder/sources/meta-fixtures/references.bb')
+  const docUri = vscode.Uri.parse(`file://${filePath}`)
+
+  suiteSetup(async function (this: Mocha.Context) {
+    this.timeout(100000)
+    const vscodeBitbake = vscode.extensions.getExtension('yocto-project.yocto-bitbake')
+    if (vscodeBitbake === undefined) {
+      assert.fail('Bitbake extension is not available')
+    }
+    await vscodeBitbake.activate()
+    await vscode.workspace.openTextDocument(docUri)
+  })
+
+  const testReferences = async (
+    position: vscode.Position,
+    // A range we hope won't break on every poky update and from which we still can assume the command worked properly
+    estimatedNumberOfReferences: { min: number, max: number },
+    nbReferencesOnOriginalDocument: vscode.Range[]
+  ): Promise<void> => {
+    let referencesResult: vscode.Location[] = []
+
+    await assertWillComeTrue(async () => {
+      referencesResult = await vscode.commands.executeCommand<vscode.Location[]>(
+        'vscode.executeReferenceProvider',
+        docUri,
+        position
+      )
+      return referencesResult.length > 0
+    }, 5000)
+    assert.equal(
+      referencesResult.length >= estimatedNumberOfReferences.min && referencesResult.length <= estimatedNumberOfReferences.max,
+      true
+    )
+    let nbReferencesFoundOnOriginalDocument = 0
+    referencesResult.forEach((reference, index) => {
+      assert.equal(reference.uri.fsPath.includes('workspaceStorage'), false)
+      if (reference.uri.fsPath === docUri.fsPath) {
+        assert.equal(
+          nbReferencesOnOriginalDocument.some((range) => range.isEqual(reference.range)),
+          true
+        )
+        nbReferencesFoundOnOriginalDocument++
+      }
+    })
+    assert.equal(nbReferencesFoundOnOriginalDocument, nbReferencesOnOriginalDocument.length)
+  }
+
+  test('References appear properly in Python on oe', async () => {
+    const position = new vscode.Position(1, 5)
+    // Got 406 on manual testing and around 600 on integration tests' environment
+    const estimatedNumberOfReferences = { min: 300, max: 1000 }
```

Comment:
this is a bit hacky. I suggest to add a fixture recipe in the fixture layer with a known reference. Ex:
```bitbake
python() {
    a = 123
    print(a)
}
```
check that print(a) has a reference to "a ="

Reply:
I first agreed and I was about to remove the test on "oe", then I realized this test is useful. 

The other bb files in the meta-fixture folder also generate "embedded language documents". When calling `vscode.executeReferenceProvider`, it gets the references of "oe" on these other "embedded language documents", which we don't want to show to the user. I spent lot of time debugging it yesterday, and I would not have noticed if it wasn't for this test.

Maybe I should "openTextDocument" on an other file, just to make sure the additional "embedded language document" is always generated. What do you think?

Reply:
The feature works well in this iteration. Users won't be that much interested in the oe references from the bitbake .py library. But oe is not the most interesting variable for references for users. I expect them to mostly want references internal function variables and bitbake variables on their .bb file and possibly includes. The current code satisfies that :)

Reply:
As for the test, if what you are testing is .bb file isolation, I suggest making it independent from the poky references. You could have another references2.bb and check that they are properly isolated references?
The goal would be to drop "estimatedNumberOfReferences" which is telling is how fragile it is..

Reply:
done